feat(pairing): enable syncing messages during local pairing

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/18895

Adds the code to sync messages when doing a local pairing.

The code is currently flagged out and the test commented. We will later add the param to let users accept syncing the messages and use that param as the condition.

jrainville committed:
feat(pairing): enable syncing messages during local pairing

Fixes https://github.com/status-im/status-desktop/issues/18895

Adds the code to sync messages when doing a local pairing.

The code is currently flagged out and the test commented. We will later add the param to let users accept syncing the messages and use that param as the condition.

jrainville commented on protocol/messenger_pairing_and_syncing.go:
```diff
@@ -22,6 +22,9 @@ type InstallationIDProvider interface {
 }
 
+// TODO use a setting or an API param instead
+const syncMessages = false
```
Will be fixed in https://github.com/status-im/status-desktop/issues/18894

jrainville commented on protocol/messenger_pairing_and_syncing.go:
```diff
@@ -450,6 +464,40 @@ func (m *Messenger) syncProfilePicturesFromDatabase(rawMessageHandler RawMessage
+	}
+
+	// Sync messages in batches of 100
+	batchSize := 100
```
Arbitrary number. Let me know if you have a better number.

jrainville commented on protocol/protobuf/messenger_local_backup.proto:
Moved the BackedUpMessage proto to pairing.proto since it's used both for local backup and pairing.

jrainville commented on protocol/protobuf/pairing.proto:
```diff
@@ -457,3 +459,49 @@ message SyncCollectiblePreferences {
+  string pinned_by = 15;
+}
+
+message BackedUpMessageBatch {
+  repeated BackedUpMessage messages = 1;
+}
```
The only new thing related to the proto

jrainville commented on tests-functional/tests/test_local_pairing.py:
```diff
@@ -212,6 +218,19 @@ def test_pairing_server_as_sender(self):
+        #     if message["id"] == message_id and message["text"] == "test_message":
+        #         message_found = True
+        #         break
+        # assert message_found, "Message sent before pairing not found on paired device"
```
This test works. If you want to test, set `syncMessages` to true and uncomment

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (15)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6948/1/) | 2025-09-23 17:38:49 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250923-173525-4f8be3-pr6948.zip) |
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6948/1/) | 2025-09-23 17:39:24 | ~4 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250923-173521-4f8be3-pr6948.zip) |
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6948/1/) | 2025-09-23 17:41:38 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250923-173531-4f8be3-pr6948.zip) |
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/1/) | 2025-09-23 17:45:40 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/1//consoleText) |
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6948/1/) | 2025-09-23 17:46:41 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250923-173523-4f8be3-pr6948.zip) |
| :heavy_check_mark: | 4f8be3fa | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/1/) | 2025-09-23 18:01:27 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6948/2/) | 2025-09-29 19:50:53 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194731-1c41b6-pr6948.zip) |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6948/2/) | 2025-09-29 19:53:33 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194739-1c41b6-pr6948.zip) |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6948/2/) | 2025-09-29 20:04:22 | ~16 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194727-1c41b6-pr6948.zip) |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6948/2/) | 2025-09-29 20:13:43 | ~26 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250929-194728-1c41b6-pr6948.zip) |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/2/) | 2025-09-29 20:14:21 | ~26 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/2//consoleText) |
| :heavy_check_mark: | 1c41b655 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/2/) | 2025-09-29 20:32:07 | ~44 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 42137fec | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6948/3/) | 2025-10-01 13:43:29 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134008-42137f-pr6948.zip) |
| :heavy_check_mark: | 42137fec | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6948/3/) | 2025-10-01 13:44:02 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134003-42137f-pr6948.zip) |
| :heavy_check_mark: | 42137fec | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6948/3/) | 2025-10-01 13:45:03 | ~4 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134017-42137f-pr6948.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6948/4/) | 2025-10-01 13:46:37 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134335-9f5a02-pr6948.zip) |
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6948/4/) | 2025-10-01 13:49:47 | ~4 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134510-9f5a02-pr6948.zip) |
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6948/4/) | 2025-10-01 13:53:12 | ~9 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-134409-9f5a02-pr6948.zip) |
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6948/4/) | 2025-10-01 13:53:28 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251001-134131-9f5a02-pr6948.zip) |
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/4/) | 2025-10-01 13:53:44 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/4//consoleText) |
| :heavy_check_mark: | 9f5a027a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/4/) | 2025-10-01 14:11:21 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6948/5/) | 2025-10-01 14:32:59 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-142937-3f10a5-pr6948.zip) |
| :heavy_check_mark: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6948/5/) | 2025-10-01 14:33:18 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-142932-3f10a5-pr6948.zip) |
| :heavy_check_mark: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6948/5/) | 2025-10-01 14:34:24 | ~4 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-142942-3f10a5-pr6948.zip) |
| :heavy_check_mark: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6948/5/) | 2025-10-01 14:41:08 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251001-142932-3f10a5-pr6948.zip) |
| :heavy_check_mark: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/5/) | 2025-10-01 14:42:52 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6948/5//consoleText) |
| :heavy_multiplication_x: | 3f10a531 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/5/) | 2025-10-01 14:56:36 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/5//consoleText) |
| :heavy_check_mark: | 3f10a531 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/6/) | 2025-10-01 15:28:24 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6948/6//consoleText) |

jrainville committed:
feat(pairing): add config to let users enable message syncing

Fixes https://github.com/status-im/status-desktop/issues/18894

Adds a param for users to enable message syncing during local pairing.

Updates the test to use that param

jrainville committed:
fix(pairing): fix pairing messages using wrong chat ID for 1-1

There was an issue where 1-1 chat messages didn't seem to be synced and it was because we were using the ChatID instead of the LocalChatID, meaning that messages from other people had our own chat ID instead of having their chat ID.

Also upgrades the integration and functional test to validate the behaviour.

jrainville committed:
fix(backup): add missing clock on backup messages

jrainville commented:
@osmaczko friendly reminder to review

jrainville committed:
test(backup): improve tests by using a map instead of a loop

dlipicar commented on protocol/messenger_pairing_and_syncing.go:
```diff
@@ -450,6 +464,40 @@ func (m *Messenger) syncProfilePicturesFromDatabase(rawMessageHandler RawMessage
+
+	// Sync messages in batches of 100
+	batchSize := 100
+	for i := 0; i < len(backupMessages); i += batchSize {
```
https://pkg.go.dev/slices#Chunk ?

jrainville replied:
Are you one of those spam bots that opens PRs on status-go?? :monocle_face:
